runtime/ms-rest/index.d.ts

```diff
@@ -0,0 +1,302 @@
+   *  - ContentType must be provided with the key name as 'Content-Type'. Default value 'application/json; charset=utf-8'.
+   *  - 'Transfer-Encoding' is set to 'chunked' by default if 'options.bodyIsStream' is set to true.
+   *  - 'Content-Type' is set to 'application/octet-stream' by default if 'options.bodyIsStream' is set to true.
+   *  - 'accept-language' by default is set to 'en-US'
+   *  - 'x-ms-client-request-id' by default is set to a new Guid. To not generate a guid for the request, please set options.disableClientRequestId to true
+   *
+   * @param {boolean} [options.disableClientRequestId] When set to true, instructs the client to not set 'x-ms-client-request-id' header to a new Guid().
+   *
+   * @param {object|string|boolean|array|number|null|undefined} [options.body] - The request body. It can be of any type. This method will JSON.stringify() the request body.
+   *
+   * @param {boolean} [options.disableJsonStringifyOnBody] - Indicates whether this method should JSON.stringify() the request body. Default value: false.
+   *
+   * @param {boolean} [options.bodyIsStream] - Indicates whether the request body is a stream (useful for file upload scenarios).
+   *
+   * @returns {object} WebResource Returns the prepared WebResource (HTTP Request) object that needs to be given to the request pipeline.
+   */
+  prepare(options?: PathTemplateBasedRequestPrepareOptions | UrlBasedRequestPrepareOptions) : WebResource;
+}
+
+/**
+ * Defines the ServiceClientCredentials. It is the base interface which enforces that the signRequest method needs to be implemented.
+ *
+ * @property {string} token               The token.
+ * @property {string} [authorizationScheme] The authorization scheme. If not specified, the default of 'Bearer" is used.
+ */
+export interface ServiceClientCredentials {
+  /**
+   * Signs a request with the Authentication header.
+   *
+   * @param {WebResource} The WebResource to be signed.
+   * @param {function(error)}  callback  The callback function.
+   */
+  signRequest(webResource: WebResource, callback: { (err: Error): void }): void;
+}
+
+/**
+ * Defines the TokenCredentials.
+ *
+ * @property {string} token               The token.
+ * @property {string} [authorizationScheme] The authorization scheme. If not specified, the default of 'Bearer" is used.
+ */
+export interface TokenCredentials extends ServiceClientCredentials {
+  token: string,
+  authorizationScheme?: string
+}
+
+/**
+ * Defines the BasicAuthenticationCredentials.
+ *
+ * @property {string} userName                 User name.
+ * @property {string} password                 Password.
+ * @property {string} [authorizationScheme]    The authorization scheme. Default ('Basic')
+ */
+export interface BasicAuthenticationCredentials extends ServiceClientCredentials {
+  userName: string,
+  password: string,
+  authorizationScheme?: string
+}
```

Comment:
include this in a small typescript app and ensure `tsc.exe` doesn't report any errors